providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/in_process.py

```diff
@@ -0,0 +1,120 @@
+        self.flag_store.shutdown()
+
+    def resolve_boolean_details(
+        self,
+        key: str,
+        default_value: bool,
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[bool]:
+        return self._resolve(key, FlagType.BOOLEAN, default_value, evaluation_context)
+
+    def resolve_string_details(
+        self,
+        key: str,
+        default_value: str,
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[str]:
+        return self._resolve(key, FlagType.STRING, default_value, evaluation_context)
+
+    def resolve_float_details(
+        self,
+        key: str,
+        default_value: float,
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[float]:
+        return self._resolve(key, FlagType.FLOAT, default_value, evaluation_context)
+
+    def resolve_integer_details(
+        self,
+        key: str,
+        default_value: int,
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[int]:
+        return self._resolve(key, FlagType.INTEGER, default_value, evaluation_context)
+
+    def resolve_object_details(
+        self,
+        key: str,
+        default_value: typing.Union[dict, list],
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[typing.Union[dict, list]]:
+        return self._resolve(key, FlagType.OBJECT, default_value, evaluation_context)
+
+    def _resolve(
+        self,
+        key: str,
+        flag_type: FlagType,
+        default_value: T,
+        evaluation_context: typing.Optional[EvaluationContext] = None,
+    ) -> FlagResolutionDetails[T]:
+        flag = self.flag_store.get_flag(key)
+        if not flag:
+            raise FlagNotFoundError(f"Flag with key {key} not present in flag store.")
+        if flag["state"] != "ENABLED":
+            return FlagResolutionDetails(default_value, reason=Reason.DISABLED)
+
+        variants = flag["variants"]
+        default = variants.get(flag.get("defaultVariant"), default_value)
+        if "targeting" not in flag or not flag["targeting"]:
+            return FlagResolutionDetails(default, reason=Reason.STATIC)
+
+        json_logic_context = evaluation_context.attributes if evaluation_context else {}
+        json_logic_context["$flagd"] = {"flagKey": key, "timestamp": int(time.time())}
+        json_logic_context["targetingKey"] = (
+            evaluation_context.targeting_key if evaluation_context else None
+        )
+        variant = jsonLogic(flag["targeting"], json_logic_context, self.OPERATORS)
+        if variant is None:
+            return FlagResolutionDetails(default, reason=Reason.DEFAULT)
+        if isinstance(variant, bool):
+            variant = str(variant).lower()
+
+        value = flag["variants"].get(variant)
+        # TODO: Check type matches
```

Comment:
Please check the type of the value. If it doesn't match, throw a type mismatch error.

Reply:
Looks like the openfeature client does this already: https://github.com/open-feature/python-sdk/blob/main/openfeature/client.py#L434

Should we still add an explicit check to the provider?